src/helpers/matchers/matchTextContent.ts

```diff
@@ -1,15 +1,23 @@
-import { Text } from 'react-native';
+import { ElementType } from 'react';
 import type { ReactTestInstance } from 'react-test-renderer';
 import { matches, TextMatch, TextMatchOptions } from '../../matches';
 import { filterNodeByType } from '../filterNodeByType';
 import { getTextContent } from '../getTextContent';
 
-export function matchTextContent(
-  node: ReactTestInstance,
-  text: TextMatch,
-  options: TextMatchOptions = {}
-) {
-  if (!filterNodeByType(node, Text)) {
+type MatchTextContentParams = {
+  node: ReactTestInstance;
+  text: TextMatch;
+  typeToMatch: ElementType | string;
+  options?: TextMatchOptions;
+};
+
+export function matchTextContent({
+  node,
+  text,
+  typeToMatch,
+  options = {},
+}: MatchTextContentParams) {
+  if (!filterNodeByType(node, typeToMatch)) {
```

Comment:
Can the matcher get it from `getConfig()`?

Reply:
This should use `getConfig().hostComponentNames.text` when we allow breaking changes but Text component when it's not the case and this is handled in text.ts file so I feel like it would be redundant to have the same checks in `matchTextContent`. Once we prepare for the next major it will always use `getConfig().hostComponentNames.text` and from this point I think it will be a good refactor but I wouldn't make the change now, wdyt ?

Reply:
I would rather go back to the previous positional signature for consistency as all matchers use that pattern now. 

re passing text type vs calculating it: I think it would be better to calculate text component type inside this method as well, even at the cost of little duplication. It will make code cleanup on release easier, as refactors will be contained in single files vs tracing signatures across different files.

Reply:
I made the change so that typeToMatch is computed within matchTextContent